<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerPathReconstructionModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerPathReconstructionModuleWidget">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>500</width>
    <height>845</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Path Reconstruction</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QGridLayout" name="ParameterLayout">
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>7</number>
     </property>
     <item row="1" column="1">
      <widget class="qMRMLNodeComboBox" name="SamplingTransformComboBox">
       <property name="toolTip">
        <string>The origin of this transform will be sampled to collect points</string>
       </property>
       <property name="nodeTypes">
        <stringlist>
         <string>vtkMRMLLinearTransformNode</string>
        </stringlist>
       </property>
       <property name="noneEnabled">
        <bool>true</bool>
       </property>
       <property name="renameEnabled">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="SamplingTransformLabel">
       <property name="text">
        <string>Sampling transform:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="ParameterNodeLabel">
       <property name="text">
        <string>Parameter node:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="qMRMLNodeComboBox" name="ParameterNodeComboBox">
       <property name="nodeTypes">
        <stringlist>
         <string>vtkMRMLPathReconstructionNode</string>
        </stringlist>
       </property>
       <property name="renameEnabled">
        <bool>true</bool>
       </property>
       <property name="selectNodeUponCreation">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QGridLayout" name="RecordingButtonLayout">
     <item row="0" column="0">
      <widget class="ctkPushButton" name="RecordingButton">
       <property name="text">
        <string>Start Recording</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QToolButton" name="DeleteButton">
       <property name="toolTip">
        <string>Delete the last path and points (hold click to see the option delete all)</string>
       </property>
       <property name="text">
        <string>...</string>
       </property>
       <property name="icon">
        <iconset resource="../qSlicerPathReconstructionModule.qrc">
         <normaloff>:/Icons/PathDelete.png</normaloff>:/Icons/PathDelete.png</iconset>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="ctkCollapsibleGroupBox" name="PathsTableGroupBox">
     <property name="title">
      <string>Paths</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="qSlicerPathReconstructionTableWidget" name="PathsTableWidget">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>200</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>200</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleGroupBox" name="AdvancedGroupBox">
     <property name="toolTip">
      <string extracomment="Show advanced model generation options">Show advanced model generation options</string>
     </property>
     <property name="title">
      <string>Advanced</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <property name="collapsed">
      <bool>true</bool>
     </property>
     <layout class="QGridLayout" name="AdvancedLayout">
      <item row="6" column="1">
       <widget class="QLineEdit" name="PathBaseNameLineEdit">
        <property name="toolTip">
         <string>Base name for path nodes that are created</string>
        </property>
        <property name="text">
         <string>Path</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="MarkupsToModelLabel">
        <property name="text">
         <string>Markups to model node:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="PathBaseNameLabel">
        <property name="text">
         <string>Path base name</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="CollectPointsLabel">
        <property name="text">
         <string>Collect points node:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QLineEdit" name="PointsBaseNameLineEdit">
        <property name="toolTip">
         <string>Base name for points nodes that are created</string>
        </property>
        <property name="text">
         <string>Points</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <layout class="QGridLayout" name="MarkupsToModelInputLayout">
        <item row="0" column="0">
         <widget class="qMRMLNodeComboBox" name="MarkupsToModelComboBox">
          <property name="toolTip">
           <string>This node contains the parameters for generating a model from the collected points</string>
          </property>
          <property name="nodeTypes">
           <stringlist>
            <string>vtkMRMLMarkupsToModelNode</string>
           </stringlist>
          </property>
          <property name="noneEnabled">
           <bool>true</bool>
          </property>
          <property name="renameEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="ctkColorPickerButton" name="MarkupsToModelColorPicker">
          <property name="toolTip">
           <string>Change the color of generated models</string>
          </property>
          <property name="displayColorName">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="1">
       <layout class="QGridLayout" name="CollectPointsInputLayout">
        <item row="0" column="0">
         <widget class="qMRMLNodeComboBox" name="CollectPointsComboBox">
          <property name="toolTip">
           <string>This node contains the parameters for point collection</string>
          </property>
          <property name="nodeTypes">
           <stringlist>
            <string>vtkMRMLCollectPointsNode</string>
           </stringlist>
          </property>
          <property name="noneEnabled">
           <bool>true</bool>
          </property>
          <property name="renameEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="ctkColorPickerButton" name="CollectPointsColorPicker">
          <property name="toolTip">
           <string>Change the color of collected points</string>
          </property>
          <property name="displayColorName">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="PointsBaseNameLabel">
        <property name="text">
         <string>Points base name:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="NextCountLabel">
        <property name="text">
         <string>Next count:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="AnchorTransformLabel">
        <property name="text">
         <string>Anchor transform:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="AnchorTransformComboBox">
        <property name="toolTip">
         <string>Points will be collected in this node's coordinate system.</string>
        </property>
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLLinearTransformNode</string>
         </stringlist>
        </property>
        <property name="baseName">
         <string/>
        </property>
        <property name="noneEnabled">
         <bool>true</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
        <property name="renameEnabled">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="NextCountSpinBox">
        <property name="toolTip">
         <string>The next points and path pair of models will have this number in their name</string>
        </property>
        <property name="maximum">
         <number>1000000</number>
        </property>
        <property name="value">
         <number>1</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
  <action name="ActionDeleteAll">
   <property name="icon">
    <iconset resource="../qSlicerPathReconstructionModule.qrc">
     <normaloff>:/Icons/PathDeleteAll.png</normaloff>:/Icons/PathDeleteAll.png</iconset>
   </property>
   <property name="text">
    <string>DeleteAll</string>
   </property>
   <property name="toolTip">
    <string>Delete all paths and points</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerPathReconstructionTableWidget</class>
   <extends>qSlicerWidget</extends>
   <header>qSlicerPathReconstructionTableWidget.h</header>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleGroupBox</class>
   <extends>QGroupBox</extends>
   <header>ctkCollapsibleGroupBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkColorPickerButton</class>
   <extends>QPushButton</extends>
   <header>ctkColorPickerButton.h</header>
  </customwidget>
  <customwidget>
   <class>ctkPushButton</class>
   <extends>QPushButton</extends>
   <header>ctkPushButton.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../qSlicerPathReconstructionModule.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>qSlicerPathReconstructionModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ParameterNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>414</x>
     <y>9</y>
    </hint>
    <hint type="destinationlabel">
     <x>412</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerPathReconstructionModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>MarkupsToModelComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>406</x>
     <y>409</y>
    </hint>
    <hint type="destinationlabel">
     <x>430</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerPathReconstructionModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>AnchorTransformComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>304</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerPathReconstructionModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>SamplingTransformComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>422</y>
    </hint>
    <hint type="destinationlabel">
     <x>314</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerPathReconstructionModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>CollectPointsComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>249</x>
     <y>422</y>
    </hint>
    <hint type="destinationlabel">
     <x>311</x>
     <y>211</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
